plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/helpers.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { InputError } from '@backstage/errors';
+import {
+  DefaultGithubCredentialsProvider,
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { OctokitOptions } from '@octokit/core/dist-types/types';
+import { parseRepoUrl } from '../publish/util';
+
+interface GetOctokitOptionsInput {
+  integrations: ScmIntegrationRegistry;
+  credentialsProvider?: GithubCredentialsProvider;
+  token?: string;
+  repoUrl: string;
+}
+export const getOctokitOptions = async ({
```

Comment:
since this isn't exported, how about just going

```ts
export async function getOctokitOptions(options: { /* inlined */ })
```

and then unpacking it in the function for bonus points :)